plugins/pipelines/create.js

```diff
@@ -18,38 +20,53 @@ module.exports = (datastore) => ({
+            const User = new Model.User(datastore, password);
 
-            /* eslint-disable consistent-return */
-            Pipeline.get(pipelineId, (error, data) => {
-                if (error) {
-                    return reply(boom.wrap(error));
-                }
-                if (data) {
-                    return reply(boom.conflict('scmUrl needs to be unique'));
-                }
-
-                Pipeline.create(request.payload, (err, result) => {
-                    if (err) {
-                        return reply(boom.wrap(err));
+            async.waterfall([
+                (next) => User.getPermissions({
+                    username,
+                    scmUrl
+                }, next),
+                (permissions, next) => {
+                    if (!permissions.admin) {
+                        return reply(boom.unauthorized(`User ${username}
+                            is not an admin of this repo`));
                     }
 
-                    const location = urlLib.format({
-                        host: request.headers.host,
-                        port: request.headers.port,
-                        protocol: request.server.info.protocol,
-                        pathname: `${request.path}/${result.id}`
-                    });
+                    return Pipeline.get(pipelineId, next);
+                },
+                (data, next) => {
+                    if (data) {
+                        next(new Error('scmUrl needs to be unique'));
+                    }
+                    const pipelineConfig = hoek.merge(request.payload,
+                        {
+                            admins:
+                            {
+                                username: true
+                            }
+                        });
 
-                    Pipeline.sync({
-                        scmUrl
-                    }, (e) => {
-                        if (e) {
-                            return reply(boom.wrap(e));
-                        }
+                    return Pipeline.create(pipelineConfig, next);
+                },
+                (pipeline, next) => Pipeline.sync({ scmUrl }, next)
+            ], (err, result) => {
+                if (err) {
+                    if (err.message === 'scmUrl needs to be unique') {
+                        return reply(boom.conflict(err));
+                    }
 
-                        return reply(result).header('Location', location).code(201);
-                    });
+                    return reply(boom.wrap(err));
+                }
+                const location = urlLib.format({
+                    host: request.headers.host,
+                    port: request.headers.port,
+                    protocol: request.server.info.protocol,
+                    pathname: `${request.path}/${result.id}`
                 });
+
+                return reply(result).header('Location', location).code(201);
```

Comment:
This `result` would be the `main` job instead of the new pipeline because sync returns the job.